### PR TITLE
MF-22: Implementing toggling password visibility

### PR DESCRIPTION
**Pull request by joeldenning**
See https://issues.openmrs.org/browse/MF-22

omrs-login-show-pass

### CODE REVIEW

File: src/login/login.component.tsx [hunk 1]
```diff
@@ -25,6 +28,12 @@ export default function Login(props: LoginProps) {
       "https://implementation-assets.sfo2.digitaloceanspaces.com/openmrs_logo_white_large.png"
   };
 
+  React.useEffect(() => {
+    if (document.activeElement !== usernameInputRef.current) {
+      passwordInputRef.current.focus();
```
Comment (joeldenning) [inline src/login/login.component.tsx]: After they click on the visibility icon button, we focus the password input so they can keep typing

File: src/login/login.component.tsx [hunk 2]
```diff
@@ -84,23 +93,50 @@ export default function Login(props: LoginProps) {
...
-                type="password"
+                type={showPassword ? "text" : "password"}
                 name="password"
                 value={password}
                 onChange={evt => setPassword(evt.target.value)}
                 placeholder="Password"
+                ref={passwordInputRef}
               />
             </label>
+            <button
```
Comment (joeldenning) [inline src/login/login.component.tsx]: An example of an icon button with an "unstyled" button. Unstyled buttons don't have a hover state -- we'll likely for other button icons want to introduce a button class specifically that gives the icon hover state.

File: src/login/login.component.tsx [hunk 3]
```diff
@@ -25,6 +28,12 @@ export default function Login(props: LoginProps) {
...
 
+  React.useEffect(() => {
+    if (document.activeElement !== usernameInputRef.current) {
+      passwordInputRef.current.focus();
+    }
+  }, [showPassword, passwordInputRef.current, usernameInputRef.current]);
+
   const input = css`
     width: 100%;
     padding: 10px 5px;
```
Comment (jonathandick) [inline src/login/login.component.tsx]: Why do use px here? In most other cases, you seem to recommend using REM.

File: src/login/login.component.tsx [hunk 4]
```diff
@@ -84,23 +93,50 @@ export default function Login(props: LoginProps) {
...
+              onClick={() => setShowPassword(!showPassword)}
+              css={css`
+                cursor: pointer;
+              `}
+            >
+              <svg
+                className="omrs-icon"
+                role="img"
+                css={css`
+                  top: calc(50% - 0.75rem);
```
Comment (jonathandick) [inline src/login/login.component.tsx]: How did you calculate the appropriate values for top and right?

File: src/login/login.component.tsx (hunk 3 above)
Comment (joeldenning) [reply src/login/login.component.tsx]: I didn't change this line of code -- it existed from when fatma and i implemented this a couple months ago.

Yes, we should be using rems, though.

File: src/login/login.component.tsx (hunk 4 above)
Comment (joeldenning) [reply src/login/login.component.tsx]: For top, I wanted it vertically centered and the total height was 1.5rem. For right, I looked at Figma and saw that there were 12px (.75rem) padding between the edge of the input and the icon.

**Pull request merged**